google/resource_storage_bucket.go

```diff
@@ -382,3 +453,130 @@ func flattenCors(corsRules []*storage.BucketCors) []map[string]interface{} {
 	}
 	return corsRulesSchema
 }
+
+func resourceGCSBucketLifecycleCreateOrUpdate(d *schema.ResourceData, sb *storage.Bucket) error {
+	if v, ok := d.GetOk("lifecycle_rule"); ok {
+		lifecycle_rules := v.([]interface{})
+
+		len_lifecycle_rules := len(lifecycle_rules)
+		if len_lifecycle_rules > 100 {
```

Comment:
I think you can use MaxItems on the resource for this and it will catch this before apply time.